packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -332,48 +321,38 @@ export class ProfilesUtils {
                 await this.promptAndHandleMissingCredentialManager(credentialManagerMap);
             }
             if (credentialManagerMap && isVSCodeCredentialPluginInstalled) {
-                return this.setupCustomCredentialManager(credentialManagerMap);
+                return new imperative.ProfileInfo("zowe", {
+                    credMgrOverride: await this.setupCustomCredentialManager(credentialManagerMap),
+                });
             }
-            return this.setupDefaultCredentialManager();
         }
 
-        const profileInfo = new imperative.ProfileInfo("zowe", {});
-        const workspacePath = ZoweVsCodeExtension.workspaceRoot?.uri.fsPath;
-        // Trigger initialize() function of credential manager to throw an error early if failed to load
-        await profileInfo.readProfilesFromDisk({
-            homeDir: FileManagement.getZoweDir(),
-            projectDir: workspacePath ? FileManagement.getFullPath(workspacePath) : undefined,
+        return new imperative.ProfileInfo("zowe", {
+            credMgrOverride: hasSecureCredentialManagerEnabled && (await this.setupDefaultCredentialManager()),
         });
-        return profileInfo;
     }
 
     public static async readConfigFromDisk(warnForMissingSchema?: boolean): Promise<void> {
         ZoweLogger.trace("ProfilesUtils.readConfigFromDisk called.");
         let rootPath: string;
-        const mProfileInfo = await ProfilesUtils.getProfileInfo();
+        this.mProfileInfo = await ProfilesUtils.setupProfileInfo();
```

Comment:
Should we use the name of the class (`ProfileUtils.`) instead of `this.` throughout static functions?